tests/plugins/subcommands/doctor/test_health_checks.py

```diff
@@ -424,3 +425,35 @@ def test_pinned_will_formatted_check(
 
         out, _, _ = conda_cli("doctor", "--verbose", "--prefix", prefix)
         assert expected_output in out
+
+
+@pytest.mark.parametrize("no_lock_flag", [True, False])
+def test_file_locking_supported(
+    tmp_env: TmpEnvFixture,
+    conda_cli: CondaCLIFixture,
+    no_lock_flag: bool,
+):
+    with tmp_env() as prefix:
+        assert _lock_impl != _lock_noop
+
+        out, _, _ = conda_cli(
+            f"CONDA_NO_LOCK={no_lock_flag}", "doctor", "--prefix", prefix
+        )
```

Comment:
this is not how we set context values, we need to use `monkeypatch.setenv` and `reset_context`:

```suggestion
        monkeypatch.setenv("CONDA_NO_LOCK", no_lock_flag)
        reset_context()

        out, _, _ = conda_cli("doctor", "--prefix", prefix)
```